### DISCUSSION TITLE
How do I create a nested span?

### DISCUSSION THREAD

**Discussion by ramgdev** [open-telemetry/opentelemetry-rust#1525] [Q&A] [answered]
Using open telemetry::trace::get_active_span, I get the parent span. But I don't see any way to create a child span off of it.

**Comment by lalitb** [comment, marked as answer]
@ramgdev - You can find the example as `rustdoc` here - https://github.com/open-telemetry/opentelemetry-rust/blob/ff64e214064df438a09a5c1695ec73866925f981/opentelemetry/src/trace/tracer.rs#L21

Also, the [server ](https://github.com/open-telemetry/opentelemetry-rust/blob/main/examples/tracing-http-propagator/src/server.rs) part of tracing-http-propagator example demonstrate two ways to create child span:
- Within the same thread through `Tracer::start_with_context()`
- Within async code through `Futures::with_context()`

**Reply by ramgdev** [reply]
Thanks for the quick reply. It works. However, is this only useful when we are tracing with context propagation? Or is there another way to establish parent/child relations between spans? Thanks again.

**Reply by lalitb** [reply]
Not without having the Context containing the parent span. So you need to use something like this:

```rust
   let parent_cx = Context::current_with_span(parent_span);
   let mut child = tracer.start_with_context("bar", &parent_cx);
```

This is in accordance with the [API Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-creation):

``
The parent Context or an indication that the new Span should be a root Span. The API MAY also have an option for implicitly using the current Context as parent as a default behavior. This API MUST NOT accept a Span or SpanContext as parent, only a full Context.
``
